src/transform/transform.ts

```diff
@@ -72,15 +72,22 @@ async function process(session: Session<CodeModel>) {
         // fix up discriminator value to use the enum type if available
         const discriminatorEnums = getDiscriminatorEnums(rootDiscriminator);
         // for each child type in the hierarchy, fix up the discriminator value
-        for (const child of values(rootDiscriminator.children?.all)) {
-          const asObj = <ObjectSchema>child;
-          let discValue = getEnumForDiscriminatorValue(asObj.discriminatorValue!, discriminatorEnums);
+        if (obj.discriminatorValue) {
+          let discValue = getEnumForDiscriminatorValue(obj.discriminatorValue!, discriminatorEnums);
           if (!discValue) {
             // this happens when there are intermediate types in the hierarchy that aren't discriminators.
             // e.g. BaseType -> GenericProps -> DerivedType
             // here, GenericProps is just a collection of standard properties common to various objects.
             // M4 gives it a discriminatorValue of "GenericProps" that won't exist in the enum defined
             // in BaseType.  unclear if this is desirable, but for now we just embed that value directly.
```

Comment:
Turns out this is a bug in M4.  We will be able to remove this in a future version.

Reply:
M4 has been fixed and incorporated.